builder/amazon/common/step_run_spot_instance.go

```diff
@@ -324,6 +324,9 @@ func (s *StepRunSpotInstance) Run(ctx context.Context, state multistep.StateBag)
 		describeOutput, err = ec2conn.DescribeInstances(&ec2.DescribeInstancesInput{
 			InstanceIds: []*string{aws.String(instanceId)},
 		})
+		if len(describeOutput.Reservations) > 0 && len(describeOutput.Reservations[0].Instances) > 0 && len(describeOutput.Reservations[0].Instances[0].BlockDeviceMappings) == 0 {
```

Comment:
Is it possible a case where not having `BlockDeviceMappings` is valid?

Reply:
Most of our builders use EBS volumes, which must make the root volume appear in the BlockDeviceMappings.

The only one I'm not sure is the instance one, as I've never actually used it.

Reply:
I notice the rest of the code behaves like it is possible to have no BlockDeviceMappings and researched a bit to understand better how it works for `instance`. It turns out that for this one it is indeed possible to not have BlockDeviceMappings, this way this needs to be workout because it can't return an error at the end. 

I was thinking you can control the retry for this case to make sure that, for example, if it retries N times and still there are no BlockDeviceMappings, then don't return the error and let the rest of the code deal with it.

Reply:
You can change the if condition to include `s.LaunchMappings != nil` because this will ensure that some BlockDeviceMappings is expected here. What do you think?